### ISSUE TITLE
Network error when checkoutAPIClient.createCardToken(...) called and Internet connectivity available

### ISSUE THREAD

**Issue by kevinpalser** [checkout/frames-ios#212]
**Describe the bug**
The Frames SDK is sometimes returning “The operation couldn’t be completed. (Frames.NetworkError error 0.)” when we know that there is Internet connectivity. This is for the checkoutAPIClient.createCardToken(card: cardTokenRequest) method. The error comes from error.localizedDescription when the result is a failure. 

One customer reported the error occurred on two separate days over a period of about 15 minutes until he was successful. We don't have many details of their user experience and how he may have varied billing data in each attempt. 

**To Reproduce**
We have not been able to reduce in our development apps. These are reports in live apps. We know that they have Internet access because other endpoints in our API are consistently successfully called immediately prior to this error.

**Expected behavior**
When the client has an Internet connection:  return a successful response with a card token or a more detailed error descriptio.  

**Smartphone (please complete the following information):**
 - Device: iPhone11 
 - OS: 15.4.1.
- Native App using Frames 3.5.3 

**Additional context**
We were told by manolis.kapetanakis@checkout.com to submit this issue ticket. We cannot log billing details used in the cardTokenRequest event systems for compliancy reasons. We need a list of what circumstances this error can be returned.

**Comment by harry-brown-cko** [comment]
Hi Kevin,

We're sorry to hear that you're experiencing issues with our SDK. We can see from our end that for your app you are using our SDK in headless mode and which errors are being raised, those being `phone_number_invalid`, `phone_number_length_invalid` and `cvv_invalid`.

`Frames.NetworkError error 0` is a validation error returned from our backend. Note that in headless mode we currently do not validate input data before making the token request. We recommend these steps to avoid these errors in future:

### Phone number validation

The tokens API has strict limits on phone number lengths. [Please see here](https://api-reference.checkout.com/#operation/requestAToken!path=2/phone&t=request)

We use PhoneNumberKit to validate phone numbers on our prebuilt UI, but only basic validation is required.

### CVV Validation

We recommend using `Frames.CardUtils.isValid(cvv: String, cardType: CardType)` to validate CVVs before making calls to `Frames.CheckoutAPIClient.createCardToken()`. 

### Reading errors after token request

It is also possible to react to errors returned from our SDK and prompt the user to fix invalid fields. Please see below:

```swift
checkoutApiClient.createCardToken(card: card) { result in
  if case .failure(.checkout(_, _, let errorCodes)) = result {
    for errorCode in errorCodes {
      switch errorCode {
      case "card_number_invalid":
        // handle error
      case "cvv_invalid":
        // handle error
      case "card_expired":
        // handle error
      case "payment_method_invalid":
        // handle error
      case "card_expiry_year_invalid":
        // handle error
      case "request_json_invalid":
        // handle error
      case "card_expiry_year_required":
        // handle error
      case "card_expiry_month_invalid":
        // handle error
      case "phone_number_invalid":
        // handle error
      case "phone_number_length_invalid":
        // handle error
      default:
        continue
      }
    }
  }
}
```

**Comment by kevinpalser** [comment]
Thank you for your prompt response Harry. We'll update our implementation based on your response and extend our error logging.

**Comment by ehab-al-cko** [comment]
Hi @kevinpalser, please let us know if you need more support. Thanks

**Comment by patrick-hoban-cko** [comment]
Hi @kevinpalser - closing this now, but please do let us know if there's anything else we can do to help.
